### PR TITLE
feat: add gs reaper cron job

### PR DESCRIPTION
**Pull request by ConjunctiveNormalForm**
Completes [[GS] Reaper job](https://github.com/orgs/Uniswap/projects/22/views/28?pane=issue&itemId=34655244)

Query and deletes all open orders for which deadline is more than an hour in the past

Also tested in my dev account

### CODE REVIEW

File: bin/stacks/cron-stack.ts
Comment (rileydcampbell) [inline bin/stacks/cron-stack.ts:1]: nit: change name of file/stack to `reaper-cron-stack`, think in other repos we've started using application specific stack naming

File: lib/crons/gs-reaper.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return order.orderHash
+      }
+      return []
+    })
+    log.info({ staleOrders }, `Found ${staleOrders.length} stale orders`)
+    if (staleOrders.length > 0) {
+      try {
+        await repo.deleteOrders(staleOrders)
+      } catch (e) {
+        log.error({ error: e }, 'Failed to delete stale orders')
```
Comment (rileydcampbell) [inline lib/crons/gs-reaper.ts:37]: Maybe add a metric here so we can add alarms based on cron failures.

**Review by rileydcampbell — APPROVED** [review]
Couple nits but otherwise lgtm!

File: bin/stacks/cron-stack.ts
Comment (ConjunctiveNormalForm) [reply bin/stacks/cron-stack.ts:1]: really? so we'd have one cron stack for each cron job? not sure if I agree with the approach tbh 😅

**Review by marktoda — APPROVED** [review]
lgtm

File: bin/stacks/cron-stack.ts
Comment (rileydcampbell) [reply bin/stacks/cron-stack.ts:1]: eh yeah maybe not then

Comment (rileydcampbell) [reply bin/stacks/cron-stack.ts:1]: I'm good with this

File: bin/stacks/cron-stack.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import { SERVICE_NAME } from '../constants'
+
+export interface CronStackProps extends cdk.NestedStackProps {
+  lambdaRole: aws_iam.Role
+  chatbotSNSArn?: string
+  envVars?: { [key: string]: string }
+}
+
+export class CronStack extends cdk.NestedStack {
```
Comment (rileydcampbell) [inline bin/stacks/cron-stack.ts:16]: Doesn't have to be in this PR but would add the alarms here for the `DeleteStaleOrdersError` metric.

**Review by rileydcampbell — APPROVED** [review]

**Pull request merged**